src/Surface.zig

```diff
@@ -4242,7 +4242,11 @@ pub fn performBindingAction(self: *Surface, action: input.Binding.Action) !bool
 
         .close_surface => self.close(),
 
-        .close_window => self.app.closeSurface(self),
+        .close_window => return try self.rt_app.performAction(
```

Comment:
Ah! I remember the history here. Close window was implemented for macOS only for a time because its a common macOS-ism. The handling is all done in the apprt and not here, so I just put whatever here since nothing ever triggered it. That is, until GTK added a menu item to do so. 😄